src/integrations/intercom/services/contact-getter.js

```diff
@@ -0,0 +1,42 @@
+const logger = require('../../../services/logger');
+
+const EMAIL_REGEX = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+function getContactQueryParameter(targetFieldValue) {
+  const targetField = targetFieldValue.match(EMAIL_REGEX) ? 'email' : 'external_id';
+
+  return {
+    query: {
+      field: targetField,
+      operator: '=',
+      value: targetFieldValue,
+    },
+  };
+}
+
+class ContactGetter {
+  static async getContact(intercomClient, Implementation, mappingValue, customerId) {
```

Comment:
Isn't the last argument the id of the record (whatever the collection name it is)?
```suggestion
  static async getContact(intercomClient, Implementation, mappingValue, recordId) {
```

Reply:
Just wanted to match the `Implementation.Intercom.getCustomer` function name but I think that function name might need to be cleaned one day (or more likely one friday afternoon)